pkg/controller/vsphere/reconciler.go

```diff
@@ -413,13 +448,24 @@ func clone(s *machineScope) (string, error) {
 		return "", fmt.Errorf("error getting devices %v", err)
 	}
 
+	// Create a new list of device specs for cloning the VM.
+	deviceSpecs := []types.BaseVirtualDeviceConfigSpec{}
+
+	// Only non-linked clones may expand the size of the template's disk.
+	if snapshotRef == nil {
```

Comment:
Nit: If the comment above this line is correct, I would expect this if statement to check the `diskMoveType == fullCloneDIskMoveType` rather than checking the `snapshotRef` which is kind of a side effect, WDYT? Behaviour should be functionally equivalent if I've understood this correctly?

Reply:
Right, it should be the same. I took this check from upstream https://github.com/kubernetes-sigs/cluster-api-provider-vsphere/blob/master/pkg/services/govmomi/vcenter/clone.go#L123 , maybe checking for snapshot reference is safer because the next line will fail otherwise.

Reply:
Why would the next line fail is this is not nil? I don't think snapshotRef is used in there is it?

Reply:
Sorry for the confusion, the next line doesn't fail. The clone task fails if diskSpec from  L460 is appended and SnapshotRef is nil.